engine/runtime-compiler/src/test/java/org/enso/compiler/test/mock/MockPackageRepository.java

```diff
@@ -0,0 +1,295 @@
+package org.enso.compiler.test.mock;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.enso.common.CompilationStage;
+import org.enso.compiler.PackageRepository;
+import org.enso.compiler.context.CompilerContext;
+import org.enso.compiler.context.CompilerContext.Module;
+import org.enso.editions.LibraryName;
+import org.enso.pkg.ComponentGroups;
+import org.enso.pkg.Package;
+import org.enso.pkg.PackageManager;
+import org.enso.pkg.QualifiedName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Option;
+import scala.collection.immutable.List;
+import scala.collection.immutable.ListSet;
+import scala.collection.immutable.ListSet$;
+import scala.collection.immutable.Seq;
+import scala.jdk.javaapi.CollectionConverters;
+import scala.runtime.BoxedUnit;
+import scala.util.Either;
+import scala.util.Left;
+import scala.util.Right;
+
+/**
+ * {@link PackageRepository} implementation with {@link org.apache.commons.vfs2.FileObject} as its
+ * type member.
+ */
+final class MockPackageRepository implements PackageRepository {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MockPackageRepository.class);
+  private final VirtualFileSystem vfs;
+  private final Path vfsRoot;
+  private final PackageManager<Path> pkgManager;
+  private final Map<LibraryName, Package<Path>> loadedPackages = new HashMap<>();
+  private final Map<String, CompilerContext.Module> loadedModules = new HashMap<>();
+  private Package<Path> mainProjectPkg;
+
+  private MockPackageRepository() {
+    this.vfs = VirtualFileSystem.create();
+    this.vfsRoot = vfs.getRoot();
+    this.pkgManager = new PackageManager<>(vfs);
+  }
+
+  static MockPackageRepository create() {
+    return new MockPackageRepository();
+  }
+
+  String listAllFilesInVfs() {
+    try {
+      return vfs.listAllFiles();
+    } catch (IOException e) {
+      LOGGER.error("Failed to list files in VFS", e);
+      return null;
```

Comment:
```suggestion
      return "Failed to list files in VFS";
```
just an idea to prevent NPE.

Reply:
Good idea. Done in 4dcef662252661404bd7b1c25664021e180d347e